closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -134,58 +134,64 @@ String getFeedback() {
      * cipher can be reused (with its original key and iv).
      */
     void reset() {
-        if (aadBuffer == null) {
-            aadBuffer = new ByteArrayOutputStream();
-        } else {
-            aadBuffer.reset();
-        }
+        synchronized (this) {
+            if (aadBuffer == null) {
+                aadBuffer = new ByteArrayOutputStream();
+            } else {
+                aadBuffer.reset();
+            }
 
-        if (ibuffer != null) {
-            ibuffer.reset();
-        }
+            if (ibuffer != null) {
+                ibuffer.reset();
+            }
 
-        if (ibuffer_enc != null) {
-            ibuffer_enc.reset();
+            if (ibuffer_enc != null) {
+                ibuffer_enc.reset();
+            }
         }
     }
 
     /**
      * Save the current content of this cipher.
      */
     void save() {
-        aadBufferSave =
-            ((aadBuffer == null || aadBuffer.size() == 0)?
-             null : aadBuffer.toByteArray());
+        synchronized (this) {
+            aadBufferSave =
+                ((aadBuffer == null || aadBuffer.size() == 0) ?
+                                null : aadBuffer.toByteArray());
```

Comment:
This is indented excessively (I suggested four more spaces than the previous line, not 16).

Reply:
Sorry, I read it 4 tabs. I removed the extra spaces.